### PR TITLE
WIP: Teaching the tidyverse, part 2

### PR DESCRIPTION
**Pull request by mine-cetinkaya-rundel**
This is still a work in progress.

To be posted Wednesday, 15 July.

### CODE REVIEW

File: content/blog/2020-07-15-teaching-the-tidyverse-in-2020-part-2-data-visualisation/index.Rmarkdown [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+  - mine
+photo:
+  url: https://unsplash.com/photos/oMpAz-DN-9I
+  author: Greg Rakozy
+editor_options:
+  markdown:
+    canonical: true
+---
+
+This is the second blog post in the "Teaching the Tidyverse in 2020" series. The first post was on [getting started](https://education.rstudio.com/blog/2020/07/teaching-the-tidyverse-in-2020-part-1-getting-started/), and today our focus is data visualisation. The following is a list includes new(ish) features of [ggplot2](https://ggplot2.tidyverse.org/) as well as not-so-new features that I see rarely in introductory teaching materials that I think belong there.
```
Comment (apreshill) [inline content/blog/2020-07-15-teaching-the-tidyverse-in-2020-part-2-data-visualisation/index.Rmarkdown:19]:
```suggestion
This is the second blog post in the "Teaching the Tidyverse in 2020" series. The first post was on [getting started](https://education.rstudio.com/blog/2020/07/teaching-the-tidyverse-in-2020-part-1-getting-started/), and today our focus is data visualisation. In this post, I'll highlight some new(ish) features of [ggplot2](https://ggplot2.tidyverse.org/) as well as not-so-new features that I see rarely in introductory teaching materials that I think belong there.
```

File: content/blog/2020-07-15-teaching-the-tidyverse-in-2020-part-2-data-visualisation/index.Rmarkdown [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+We'll be using the `penguins` data from the palmerpenguins package. You can read about the variables in the data frame [here](https://allisonhorst.github.io/palmerpenguins/reference/penguins.html) and a quick peek is provided below.
+
+```{r glimpse-penguins}
+glimpse(penguins)
+```
+
+## Making a single box
+
+At some point in ggplot2 history making a box plot of a single variable required adding `factor = 1` to the aesthetic mapping, which I never found to be easy to explain.
```
Comment (apreshill) [inline content/blog/2020-07-15-teaching-the-tidyverse-in-2020-part-2-data-visualisation/index.Rmarkdown:40]:
```suggestion
At some point in ggplot2 history, making a box plot of a single variable required adding `factor = 1` to the aesthetic mapping, which I never found to be easy to explain.
```

File: content/blog/2020-07-15-teaching-the-tidyverse-in-2020-part-2-data-visualisation/index.Rmarkdown [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+Horizontal bar plots can be really useful, especially for categorical data whose levels have long names that overlap if placed on the x-axis. Previously, making horizontal bar plots required mapping the variable to be plotted to the `x` aesthetic and then applying a `coord_flip()` layer to flip the axes, e.g.
+
+```{r penguins-species-bar-coord-flip}
+ggplot(penguins, aes(x = species)) +
+  geom_bar() +
+  coord_flip()
+```
+
+`geom_bar()` now works in both directions, do the categorical variable can be directly mapped to the `y` aesthetic to achieve the horizontal box plot.
```
Comment (apreshill) [inline content/blog/2020-07-15-teaching-the-tidyverse-in-2020-part-2-data-visualisation/index.Rmarkdown:64]:
```suggestion
`geom_bar()` now works in both directions, so the categorical variable can be directly mapped to the `y` aesthetic to achieve the horizontal box plot.
```

File: content/blog/2020-07-15-teaching-the-tidyverse-in-2020-part-2-data-visualisation/index.Rmarkdown [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+
+```{r penguins-species-props-bar}
+penguins %>%
+  count(species) %>%
+  mutate(prop = n / sum(n)) %>%
+  ggplot(aes(x = prop, y = species)) +
+  geom_col()
+```
+
+First, let's reorder the bars in descending order, with the penguin with the highest proportion on top. We use `fct_reorder()` function from the forcats package for this. We can read `fct_reorder(species, prop)` as "reorder the levels of `species` based on the values of `prop`".
```
Comment (apreshill) [inline content/blog/2020-07-15-teaching-the-tidyverse-in-2020-part-2-data-visualisation/index.Rmarkdown:95]:
```suggestion
First, let's reorder the bars in descending order, with the penguin species with the highest proportion on top. We use `fct_reorder()` function from the forcats package for this. We can read `fct_reorder(species, prop)` as "reorder the levels of `species` based on the values of `prop`".
```

File: content/blog/2020-07-15-teaching-the-tidyverse-in-2020-part-2-data-visualisation/index.Rmarkdown [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+
+ggplot2 is a vast package, and I don't think it's feasible to teach the complete functionality of the package within an intro stats or data science curriculum. That being said, I think it is important to not just teach how to make plots, but how to make good plots that follow data visualisation best practices. Leveraging packages like forcats and scales allows you to do that without adding too much cognitive burden on the learners.
+
+## The rest of the gg-verse
+
+There are many packages that extend ggplot2 (e.g. [gganimate](https://gganimate.com/), [ggridges](https://wilkelab.org/ggridges/), [ggrepel](https://ggrepel.slowkow.com/), [GGally](https://ggobi.github.io/ggally/)) and just like it's not possible to cover all of ggplot2 in an intro course, it's definitely not possible to cover the entire gg-verse. However it is useful to introduce your students to some of these packages just so they are aware of the existence of the larger ecosystem.
```
Comment (apreshill) [inline content/blog/2020-07-15-teaching-the-tidyverse-in-2020-part-2-data-visualisation/index.Rmarkdown:150]:
```suggestion
There are many packages that extend ggplot2 (e.g. [gganimate](https://gganimate.com/), [ggridges](https://wilkelab.org/ggridges/), [ggrepel](https://ggrepel.slowkow.com/), [GGally](https://ggobi.github.io/ggally/)) and just like it's not possible to cover all of ggplot2 in an intro course, it's definitely not possible to cover the entire gg-verse. However, it is useful to introduce your students to some of these packages so they are aware of the existence of the larger ecosystem.
```

File: content/blog/2020-07-15-teaching-the-tidyverse-in-2020-part-2-data-visualisation/index.Rmarkdown [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+
+A particularly good place to start is when you find a pain point with ggplot2 and the solution is using one of these other packages (e.g. ggrepel for labelling data points on a plot without overlapping text).
+
+Another good approach for introducing these packages are assignments where students independently explore packages you haven't explicitly taught them. I like doing this using the following two assignments:
+
+-   **You gotta pick a package or two:** Provide a list of fun and/or easy to onboard packages and ask students to read the package documentation and use the package to do "something".
```
Comment (apreshill) [inline content/blog/2020-07-15-teaching-the-tidyverse-in-2020-part-2-data-visualisation/index.Rmarkdown:156]: I'm unsure why something is in quotes here.

File: content/blog/2020-07-15-teaching-the-tidyverse-in-2020-part-2-data-visualisation/index.Rmarkdown [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+
+-   **Mirror, mirror on the wall, who's the ugliest of them all?** I got the idea for this assignment Give a simple plot and ask students to make the plot as ugly as possible. I like doing this with a disclaimer that says "*I realize that ugly is subjective, so we're mostly looking to see if you can figure out how to change the look of a plot using help files of functions you haven't learned before. As you do, try to make your plot as visually unappealing as possible.*"I got the idea for this assignment from Drew Steen, who shared [on Twitter](https://twitter.com/biogeobiochem/status/1172547837046820864) sample student plots that are, well, awful, in the best way possible.
```
Comment (apreshill) [inline content/blog/2020-07-15-teaching-the-tidyverse-in-2020-part-2-data-visualisation/index.Rmarkdown:158]:
```suggestion
-   **Mirror, mirror on the wall, who's the ugliest of them all?** Give a simple plot and ask students to make the plot as ugly as possible. I like doing this with a disclaimer that says "*I realize that ugly is subjective, so we're mostly looking to see if you can figure out how to change the look of a plot using help files of functions you haven't learned before. As you do, try to make your plot as visually unappealing as possible.*"I got the idea for this assignment from Drew Steen, who shared [on Twitter](https://twitter.com/biogeobiochem/status/1172547837046820864) sample student plots that are, well, awful, in the best way possible.
```

Comment (apreshill) [reply content/blog/2020-07-15-teaching-the-tidyverse-in-2020-part-2-data-visualisation/index.Rmarkdown:158]: Allison Horst also has a great blog post on using this for teaching: https://www.allisonhorst.com/post/do-your-worst/

Comment (mine-cetinkaya-rundel) [reply content/blog/2020-07-15-teaching-the-tidyverse-in-2020-part-2-data-visualisation/index.Rmarkdown:158]: Thanks!

**Pull request merged**